src/SmoFacade/Server.cs

```diff
@@ -15,81 +15,93 @@ namespace AgDatabaseMove.SmoFacade
+      cmd.Parameters.Add(new SqlParameter("db_name", SqlDbType.VarChar) { Value = dbName });
+      cmd.CommandText = query;
+      using var reader = cmd.ExecuteReader();
+      reader.Read();
+      return (decimal)reader["total_size_mb"];
+    }
+
+    public int RemainingDiskMb()
+    {
+      // https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-master-files-transact-sql?view=sql-server-ver15
+      var query = "SELECT DISTINCT CONVERT(INT, dovs.available_bytes/1048576.0) AS free_size_mb " +
+                  "FROM sys.master_files mf " +
+                  "CROSS APPLY sys.dm_os_volume_stats(mf.database_id, mf.FILE_ID) dovs " +
+                  "WHERE mf.type IN(0,1) " + // Rows and Logs
+                  "AND SUBSTRING(CONVERT(VARCHAR, SERVERPROPERTY('InstanceDefaultDataPath')), 1, 3) = dovs.volume_mount_point"; // Disk Drive equals
+
+      using var cmd = SqlConnection.CreateCommand();
+      cmd.CommandText = query;
+      using var reader = cmd.ExecuteReader();
+      reader.Read();
+      return (int)reader["free_size_mb"];
+    }
+
+    private static string GetFileName(string physicalName)
     {
-      SqlConnection?.Dispose();
-    }
-
-    public void DropLogin(LoginProperties login)
-    {
-      _server.Logins[login.Name].DropIfExists();
-    }
-
-    public decimal DatabaseSizeMb(string dbName)
-    {
-      var query = "SELECT total_size_mb = (SUM(size) * 8. / 1024) " +
-                  "FROM sys.master_files WITH(NOWAIT) " +
-                  "WHERE database_id = DB_ID(@db_name) " +
-                  "GROUP BY database_id";
-
-      using var cmd = SqlConnection.CreateCommand();
-      cmd.Parameters.Add(new SqlParameter("db_name", SqlDbType.VarChar) { Value = dbName });
-      cmd.CommandText = query;
-      using var reader = cmd.ExecuteReader();
-      reader.Read();
-      return (decimal)reader["total_size_mb"];
-    }
-
-    public int RemainingDiskMb()
-    {
-      // https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-master-files-transact-sql?view=sql-server-ver15
-      var query = "SELECT DISTINCT CONVERT(INT, dovs.available_bytes/1048576.0) AS free_size_mb " +
-                  "FROM sys.master_files mf " +
-                  "CROSS APPLY sys.dm_os_volume_stats(mf.database_id, mf.FILE_ID) dovs " +
-                  "WHERE mf.type IN(0,1) " + // Rows and Logs
-                  "AND SUBSTRING(CONVERT(VARCHAR, SERVERPROPERTY('InstanceDefaultDataPath')), 1, 3) = dovs.volume_mount_point"; // Disk Drive equals
-
-      using var cmd = SqlConnection.CreateCommand();
-      cmd.CommandText = query;
-      using var reader = cmd.ExecuteReader();
-      reader.Read();
-      return (int)reader["free_size_mb"];
+      if(Path.IsPathRooted(physicalName))
+        return Path.GetFileName(physicalName) ??
+               throw new InvalidBackupException($"Physical name in backup is incomplete: {physicalName}");

```

Comment:
What happens if it is not rooted?
Should we throw an exception?

Reply:
See code lines below.... chop off everything before `\\` to get filename

Reply:
Right for the windows paths, but how bout linux paths?

Reply:
I'm pretty sure `IsPathRooted` == true even on windows, but I need to test that more thoroughly